core/src/main/java/hudson/PluginManager.java

```diff
@@ -2661,7 +2662,10 @@ public boolean isActivated() {
         public Map<PluginWrapper, String> getDeprecatedPlugins() {
             return Jenkins.get().getPluginManager().getPlugins().stream()
                     .filter(PluginWrapper::isDeprecated)
-                    .collect(Collectors.toMap(Function.identity(), it -> it.getDeprecations().get(0).url));
+                    .sorted(Comparator.comparing(PluginWrapper::getDisplayName)) // Sort by plugin name
```

Comment:
should use String.CASE_INSENSITIVE in the Comparator to deal with mixed lowercase and uppercase plugin names